src/olympia/blocklist/utils.py

```diff
@@ -119,7 +119,10 @@ def disable_addon_for_block(block):
                 ver
                 for ver in block.addon_versions
                 # We don't need to reject versions from older deleted instances
-                if ver.addon == block.addon and block.is_version_blocked(ver.version)
+                # and already disabled files
+                if ver.addon == block.addon
+                and block.is_version_blocked(ver.version)
+                and ver.file.status != amo.STATUS_DISABLED
```

Comment:
Is that even possible?

Reply:
Hm, if I add ´and ver.deleted = False` in there, we wouldn't reject those, but also not update the wrong status. That does not seem desirable?

Reply:
There's no database constraint enforcing that deleted versions have disabled files 😬 
It's an edge case though (all versions that are deleted _now_ have the file status set to disabled), so it could be ignored if you're okay with that?  I.e. just assume all deleted versions also have the file set to disabled.  (what your code is assuming)

I'd lean away from updating "wrong" statuses.

Reply:
On prod there are currently 11 deleted versions with files that are approved. I checked one and it was because the version was deleted in between the auto-approval cron determining the files to be approved and the actual approval, which we fixed in #19136.

I'd leave this as is and fix the 11 versions on prod manually, if that works for you?

Reply:
wfm.